test/server/lib/DocApi.ts

```diff
@@ -2766,6 +2766,20 @@ function testDocApi() {
     });
   });
 
+  describe("Allowed Origin", () => {
+    it('should allow only example.com',  async () => {
+      async function check(origin: string, status: number, error?: string) {
+        const resp = await axios.get(`${serverUrl}/api/docs/${docIds.Timesheets}/`,
+        {...chimpy, headers: {...chimpy.headers, "Origin": origin}}
+        );
+        assert.equal(resp.status, status);
+        error && assert.deepEqual(resp.data, {error});
+      }
+      await check("https://www.toto.com", 500, "Unrecognized origin");
+      await check("https://example.com", 200);
```

Comment:
Can you try thinking of more good and bad variants - `https://foo.example.com`, `http://example.com:8080`, `https://example.com/path`, `https://bad.com/example.com`, `https://badexample.com` - hmm that last one makes me nervous

Reply:
I think the domain check needs to be more careful to avoid e.g. badexample.com. The isUrlAllowed function you drew on from webhooks has the same problem. Do you think you could factor out a function they both use, so both get fixed?

Reply:
The tests you provide made me have multiples headaches to have the code correct. Tell me if you're ok with the actual  implementation and I'll look into the webhooks.

Reply:
Wouldn't a simpler method like this address the problem well?

```
function matchesBaseDomain(domain, baseDomain) {
  return domain === baseDomain || domain.endsWith("." + baseDomain);
}
```

Reply:
I thinks it's way smarter ... thank you ! 😅